src/core_ocean/shared/mpas_ocn_vmix_gotm.F

```diff
@@ -0,0 +1,494 @@
+
+      call mpas_timer_stop('gotm cell loop')
+
+      deallocate(shearSquaredTop)
+      deallocate(NSquaredTop)
+      deallocate(hLayer)
+
+   !--------------------------------------------------------------------
+
+   end subroutine ocn_vmix_coefs_gotm_build!}}}
+
+!***********************************************************************
+!
+!  routine ocn_vmix_gotm_init
+!
+!> \brief   Initializes ocean vertical mixing quantities using GOTM
+!> \author  Qing Li
+!> \date    11 May 2020
+!> \details
+!>  This routine initializes a variety of quantities related to
+!>  vertical mixing in the ocean. Parameters are set by calling into GOTM
+!
+!-----------------------------------------------------------------------
+
+
+   subroutine ocn_vmix_gotm_init(domain,err)!{{{
+
+   !--------------------------------------------------------------------
+
+      implicit none
+
+      type (domain_type), intent(inout) :: domain !< Input/Output: domain information
+
+      integer, intent(out) :: err !< Output: error flag
+
+      type (block_type), pointer :: block
+      type (mpas_pool_type), pointer :: meshPool
+      type (mpas_pool_type), pointer :: statePool
+
+      integer, pointer :: nVertLevels, nVertLevelsP1
+      integer, dimension(:), pointer :: nCellsArray
+
+      real (kind=RKIND), dimension(:,:), pointer :: &
+        gotmVertViscTopOfCellCur, gotmVertDiffTopOfCellCur, &
+        gotmTKETopOfCellCur, gotmDissTopOfCellCur, gotmLengthTopOfCellCur, &
+        gotmKbTopOfCellCur, gotmEpsbTopOfCellCur
+
+      integer :: namlstUnit !< unit for GOTM namelist file
+      integer :: k, ki, iCell, nCells
+      type (MPAS_timeInterval_type) :: timeStep
+
+      !
+      ! assume no errors during initialization and set to 1 when error is encountered
+      !
+      err=0
+
+      gotmOn = config_use_gotm
+
+      !
+      ! only initialize if GOTM is turned on
+      !
+      if (.not. gotmOn) return
+
+      !
+      ! When GOTM is turned on, all other vertical mixing schemes should be off
+      ! Test to make sure this is the case.
+      !
+      ! test here, err=1 if a problem
+
+      ! use MPAS-O dt here. Sub-stepping is possible, though it may not help too much
+      ! without updating the shear and stratification in the substeps
+      timeStep = mpas_get_clock_timestep(domain % clock, ierr=err)
+      call mpas_get_timeInterval(timeStep, dt=gotm_dt)
+      !
+      ! pull nVertLevels out of the mesh structure
+      !
+      call mpas_pool_get_dimension(domain % blocklist % dimensions, 'nVertLevels', nVertLevels)
+
+      ! initialize GOTM parameters
+      gotmSurfaceRoughnessLength = config_gotm_constant_surface_roughness_length
+      gotmBottomRoughnessLength = config_gotm_constant_bottom_roughness_length
+      gotmBottomDragCoef = config_gotm_constant_bottom_drag_coeff
+
+      !
+      ! initialize GOTM turbulence module
+      ! read in GOTM namelist, required by init_turbulence
+      call mpas_new_unit(namlstUnit)
```

Comment:
When I tested gotm I forgot to include the gotm namelist file in the run directory. But the error messages don't indicate what is wrong. Please check here if the file exists. If it does not, call 
```
call mpas_log_write(' fatal error: gotm namelist file does not exist: config_gotm_namelist_file = '//config_gotm_namelist_file, MPAS_LOG_ERR)
```
so the user knows. Thanks!

Reply:
@mark-petersen This should be checked in GOTM. But since we are suppressing all the messages from GOTM we won't see the error message. I added a check here as suggested.